### ISSUE TITLE
Derivations with extra constraints

### ISSUE THREAD

**Issue by mlutze** [flix/flix#2394]
```scala
opaque type Thing[a] with Eq = Set[a]
```

This fails because `Set[a]` has a type constraint `Order[a]` which isn't covered by the constraints generated in the Deriver.

**Comment by magnus-madsen** [comment]
What is the status on this?

**Comment by mlutze** [comment]
The particular example is fixed, but general problem is not. It's complicated because Deriver comes before Typer, but typing has to be done before superclass resolution. So some redesign has to be done.

**Comment by mlutze** [comment]
I wasn't counting this as a priority

**Comment by magnus-madsen** [comment]
OK, its not a priority. Just wondering because I remembered that we did something :)

**Comment by magnus-madsen** [comment]
Is this an issue today?
